pkg/status/deployment_test.go

```diff
@@ -0,0 +1,112 @@
+package status
+
+import (
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/require"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetDeploymentStatusConditions(t *testing.T) {
+
+	t.Run("test deployment status conditions", func(t *testing.T) {
+		t.Run("deployment does not exist", func(t *testing.T) {
+			fakeClient := test.NewFakeClient(t)
+			conditions, err := GetDeploymentStatusConditions(fakeClient, "test-deployment", test.HostOperatorNs)
+			require.Error(t, err)
+			require.Equal(t, 1, len(conditions))
+			require.Equal(t, toolchainv1alpha1.ConditionReady, conditions[0].Type)
+			require.Equal(t, corev1.ConditionFalse, conditions[0].Status)
+			require.Equal(t, "DeploymentNotFound", conditions[0].Reason)
+			require.Contains(t, conditions[0].Message, "unable to get the deployment")
+			require.True(t, isRecent(conditions[0].LastTransitionTime), "LastTransitionTime should be recently updated")
+			require.True(t, isRecent(*conditions[0].LastUpdatedTime), "LastUpdatedTime should be recently updated")
+		})
+
+		t.Run("deployment not available", func(t *testing.T) {
+			fakeClient := test.NewFakeClient(t, fakeDeploymentNotAvailable())
+			conditions, err := GetDeploymentStatusConditions(fakeClient, "test-deployment", test.HostOperatorNs)
+			require.Error(t, err)
+			require.Equal(t, 1, len(conditions))
+			require.Equal(t, toolchainv1alpha1.ConditionReady, conditions[0].Type)
+			require.Equal(t, corev1.ConditionFalse, conditions[0].Status)
+			require.Equal(t, "DeploymentNotReady", conditions[0].Reason)
+			require.Contains(t, conditions[0].Message, "deployment has unready status conditions")
+			require.True(t, isRecent(conditions[0].LastTransitionTime), "LastTransitionTime should be recently updated")
+			require.True(t, isRecent(*conditions[0].LastUpdatedTime), "LastUpdatedTime should be recently updated")
+		})
+
+		t.Run("deployment not progressing", func(t *testing.T) {
+			fakeClient := test.NewFakeClient(t, fakeDeploymentNotProgressing())
+			conditions, err := GetDeploymentStatusConditions(fakeClient, "test-deployment", test.HostOperatorNs)
+			require.Error(t, err)
+			require.Equal(t, 1, len(conditions))
+			require.Equal(t, toolchainv1alpha1.ConditionReady, conditions[0].Type)
```

Comment:
you could add a custom `assertCondition` func to avoid repeating the same block of lines in each test (but it's a minor suggestion)

Reply:
I've included this with https://issues.redhat.com/browse/CRT-729 as well